Source/Engine/Render2D/SpriteAtlas.cpp

```diff
@@ -153,11 +159,26 @@ bool SpriteAtlas::LoadSprites(ReadStream& stream)
         return true;
     }
     stream.ReadInt32(&tilesCount);
-    Sprites.Resize(tilesCount);
-    for (Sprite& t : Sprites)
+    Sprites.EnsureCapacity(tilesCount);
+    for (int i = 0; i < tilesCount; ++i)
     {
+        Sprite t{};
+
         stream.Read(t.Area);
         stream.ReadString(&t.Name, 49);
+        if (tilesVersion >= 2)
+        {
+            stream.ReadUint32(&t.Id.Values[0]);
+            stream.ReadUint32(&t.Id.Values[1]);
+            stream.ReadUint32(&t.Id.Values[2]);
+            stream.ReadUint32(&t.Id.Values[3]);
+        }
+        else
+        {
+            t.Id = Guid::New();
```

Comment:
This should be changed according to the aforementioned missing "project upgrading" processing. Before/after creating a new id for older format `SpriteAtlas` instances, replacements should be made where needed (i.e. `SpriteRender` instances).